posthog-react-native/src/posthog-rn.ts

```diff
@@ -22,14 +26,19 @@ export type PostHogOptions = PostHogCoreOptions & {
   customAppProperties?:
     | PostHogCustomAppProperties
     | ((properties: PostHogCustomAppProperties) => PostHogCustomAppProperties)
+  /** Allows you to provide a custom and synchronous storage such as mmkv */
+  customStorage?: PostHogCustomSyncStorage
+  /** Allows you to provide a custom and asynchronous storage such as async-storage, expo-file-system or something else.
+   * By default, the SDK will try to load expo-file-system and then async-storage if none is provided.
+   */
```

Comment:
@benjackwhite the downside of having both types is that if a user passes both, `customStorage` will have precedence which is fine but it's unclear for the user, I will add as a comment

Reply:
Added a warning as well if both are set.